core/web/external_initiators_controller.go

```diff
@@ -50,8 +51,10 @@ func (eic *ExternalInitiatorsController) Destroy(c *gin.Context) {
 		return
 	}
 
-	id := c.Param("AccessKey")
-	if err := eic.App.GetStore().DeleteExternalInitiator(id); err != nil {
+	name := c.Param("Name")
+	if exi, err := eic.App.GetStore().FindExternalInitiatorByName(name); err == orm.ErrorNotFound {
+		jsonAPIError(c, http.StatusNotFound, err)
+	} else if err := eic.App.GetStore().DeleteExternalInitiator(exi.Name); err != nil {
```

Comment:
`Delete` doesn't return `orm.ErrorNotFound` if it doesn't find a match?

Reply:
No surprisingly not.  Neither does it happen for DeleteBridge(..).  I followed the same pattern used by the bridge controller here.